picky-asn1-x509/src/private_key_info.rs

```diff
@@ -130,6 +143,29 @@ impl PrivateKeyInfo {
             version: 0,
             private_key_algorithm: AlgorithmIdentifier::new_elliptic_curve(curve_oid.into()),
             private_key,
+            public_key: None,
+        }
+    }
+
+    pub fn new_ed_encryption(
+        algorithm: ObjectIdentifier,
+        secret: impl Into<OctetStringAsn1>,
+        public_key: Option<BitString>,
+    ) -> Self {
+        let secret: OctetStringAsn1 = secret.into();
+
+        let (version, public_key) = if let Some(public_key) = public_key {
+            // If the public key is present, the version MUST be set to v2(1)
+            (1, Some(ExplicitContextTag1(Optional(public_key.into()))))
+        } else {
+            (0, None)
+        };
```

Comment:
The comment is helping, but maybe you could declare `const`s to define the values once. I see this is also used at other places and it would probably help with readability. These could stay private if it’s not intended to be used by downstream code.

Reply:
Done